projects/libs/core-library-typescript/test/transaction-pool/transaction-pool.spec.ts

```diff
@@ -32,13 +28,53 @@ function aTransaction() {
 }
 
 describe("new broadcast transaction", () => {
+  let transactionPool: TransactionPool;
+  let gossip: GossipClient;
+
+  beforeEach(() => {
+    gossip = stubInterface<types.GossipClient>();
+    transactionPool = new TransactionPool(gossip);
+  });
+
   it("is added to the transaction pool", async () => {
     const tx = aTransaction();
     await transactionPool.addNewPendingTransaction(tx);
+
     const { transactions } = await transactionPool.getAllPendingTransactions();
+
     transactions.should.eql([tx]);
     expect(gossip.broadcastMessage).to.have.been.called;
   });
 
-  xit("two identical transaction are processed only once");
```

Comment:
this works only if they are committed to the same block

Reply:
True.